packages/react/src/components/Modal/ModalContext.tsx

```diff
@@ -0,0 +1,7 @@
+import { createContext } from 'react';
+
+export type ModalContextProps = {
+  closeModal?: () => void;
+};
+
+export const ModalContext = createContext<ModalContextProps | null>(null);
```

Comment:
Suggest just putting this in `Modal.tsx` since its where the `Provider` is used and its the "parent", I have done the same for several other components. This is to reduce amount of files and improve readability.